api/files/views.py

```diff
@@ -162,3 +166,85 @@ def get_serializer_context(self):
         context = JSONAPIBaseView.get_serializer_context(self)
         context['file'] = self.file
         return context
+
+
+class FileMetadataRecordsList(JSONAPIBaseView, generics.ListAPIView, FileMixin):
+
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        PermissionWithGetter(ContributorOrPublic, 'target'),
+    )
+
+    required_read_scopes = [CoreScopes.NODE_FILE_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    serializer_class = FileMetadataRecordSerializer
+    view_category = 'files'
+    view_name = 'metadata-records'
+
+    ordering = ('-created',)
+
+    def get_queryset(self):
+        return self.get_file().records.all()
+
+
+class FileMetadataRecordDetail(JSONAPIBaseView, generics.RetrieveUpdateAPIView, FileMixin):
+
+    record_lookup_url_kwarg = 'record_id'
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        PermissionWithGetter(ContributorOrPublic, 'target'),
+        PermissionWithGetter(ReadOnlyIfRegistration, 'target'),
+    )
+
+    required_read_scopes = [CoreScopes.NODE_FILE_READ]
+    required_write_scopes = [CoreScopes.NODE_FILE_WRITE]
+
+    serializer_class = FileMetadataRecordSerializer
+    view_category = 'files'
+    view_name = 'metadata-record-detail'
+
+    def get_object(self):
+        return utils.get_object_or_error(
+            self.get_file().records.filter(_id=self.kwargs[self.record_lookup_url_kwarg]),
+            request=self.request,
+        )
+
+
+class FileMetadataRecordDownload(JSONAPIBaseView, generics.RetrieveAPIView, FileMixin):
+
+    record_lookup_url_kwarg = 'record_id'
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        PermissionWithGetter(ContributorOrPublic, 'target'),
+    )
+
+    required_read_scopes = [CoreScopes.NODE_FILE_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    view_category = 'files'
+    view_name = 'metadata-record-download'
+
+    def get_serializer_class(self):
+        return None
+
+    def get_object(self):
+        return utils.get_object_or_error(
+            self.get_file().records.filter(_id=self.kwargs[self.record_lookup_url_kwarg]).select_related('schema', 'file'),
+            request=self.request,
+        )
+
+    def get(self, request, **kwargs):
+        file_type = self.request.query_params.get('export', 'json')
```

Comment:
@erinspace, FYI: I used `export` instead of `format` here because `format` is used by DRF to determine whether to return the browsable api or JSON.
http://www.django-rest-framework.org/topics/browsable-api/#formats

Reply:
We need a check here to make sure people don't try something like `?export=wombat`, would be nice to have an allowed_file_metadata_formats settings var or static member on this view.

Reply:
Yep, @caseyrollins had that [covered here](https://github.com/CenterForOpenScience/osf.io/pull/8721/files#diff-898ca0bd378d7ba622a0e1b2c78f2abfR31) -- see [this test](https://github.com/CenterForOpenScience/osf.io/pull/8721/files#diff-75927fb4e67948f8a9b55f939db47937R64)